unittests/svnn_ibc_tests.cpp

```diff
@@ -18,6 +18,46 @@ using namespace eosio::testing;
 
 using mvo = mutable_variant_object;
 
+std::string binary_to_hex(const std::string& bin) {
+   size_t pad_bin_length = (4 - (bin.length() % 4)) % 4;
+
+    std::string padded_bin = std::string(pad_bin_length, '0') + bin;
+    std::stringstream hex_stream;
+
+    for (size_t i = 0; i < padded_bin.length(); i += 4) {
+        std::string segment = padded_bin.substr(i, 4);
+        int value = std::stoi(segment, nullptr, 2);
+        hex_stream << std::hex << value;
+    }
+
+    size_t pad_out_length = (2 - (hex_stream.str().length() % 2)) % 2;
+
+    std::string padded_hex_stream = std::string(pad_out_length, '0') + hex_stream.str();
+    std::string reversed_hex = "";
+
+    for (size_t i = 0; i < padded_hex_stream.length(); i += 2) {
+      std::string byte = padded_hex_stream.substr(i, 2);
+      reversed_hex = byte + reversed_hex;
+    }
+
+    return reversed_hex;
+
+}
+
+std::string bitset_to_binary(const boost::dynamic_bitset<unsigned char>& bitset) {
+   std::string result;
+   result.reserve(bitset.size());
+   for (std::size_t i = bitset.size(); i > 0; --i) {

```

Comment:
You can start from `bitset.size() - 1`. Then you can use `bitset[i]` instead of `bitset[i - 1]` below.

Will `bitset` never be empty?

Reply:
But then you have to be careful to not use `i >= 0` which ia always true for an unsigned like `std::size_t`